File: fe-tienda/src/app/core/models/user-dto.ts
export interface UserDTO {
    id: number;
    username: string;
    firstName: string;
    lastName: string;
    country: string;
    role: string;
    email: string;
    accountActivated: boolean;
}

export interface UpdateUserDTO {
    firstName: string;
    lastName: string;
    country: string;
    email: string;
}

File: bff-tienda/src/main/java/com/example/tiendabackend/dto/UserDTO.java
package com.example.tiendabackend.dto;

import lombok.Data;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Data
public class UserDTO {
    @NotNull
    private Long id;
    @NotBlank
    private String username;
    @NotBlank
    private String firstName;
    @NotBlank
    private String lastName;
    @NotBlank
    private String country;
    @NotBlank
    private String role;
    @Email
    private String email;
    private boolean accountActivated;
}
